executor.go

```diff
@@ -176,23 +195,49 @@ func (e *Executor) verifySubjectAgainstReferrers(ctx context.Context, task *exec
 		}
 		return nil
 	})
-	if err != nil {
+	if err != nil && err != errSubjectPruned {
 		return nil, fmt.Errorf("failed to verify referrers for artifact %s: %w", artifact, err)
 	}
```

Comment:
Consider
```suggestion
	if err != nil {
		if err != errSubjectPruned {
			return nil, fmt.Errorf("failed to verify referrers for artifact %s: %w", artifact, err)
		}
		newTasks = nil
	}
```
and remove line 208-210.